From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect eswitch.h include
Patch-mainline: never, kabi
References: kabi

patches.suse/net-mlx5-E-Switch-Avoid-setup-attempt-if-not-being-e.patch

In networking-stable-18_07_19, commit
0efc8562491b7d36f6bbc4fbc8f3348cb6641e9c (net/mlx5: E-Switch, Avoid
setup attempt if not being e-switch manager) added few #includes to a
couple of files.  It made the kABI checker to complain.

Just hide the includes from the kABI checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>

---
 drivers/net/ethernet/mellanox/mlx5/core/fs_core.c |    2 ++
 drivers/net/ethernet/mellanox/mlx5/core/fw.c      |    2 ++
 2 files changed, 4 insertions(+)

--- a/drivers/net/ethernet/mellanox/mlx5/core/fs_core.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/fs_core.c
@@ -36,7 +36,9 @@
 #include "mlx5_core.h"
 #include "fs_core.h"
 #include "fs_cmd.h"
+#ifndef __GENKSYMS__
 #include "eswitch.h"
+#endif
 #include "diag/fs_tracepoint.h"
 
 #define INIT_TREE_NODE_ARRAY_SIZE(...)	(sizeof((struct init_tree_node[]){__VA_ARGS__}) /\
--- a/drivers/net/ethernet/mellanox/mlx5/core/fw.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/fw.c
@@ -34,7 +34,9 @@
 #include <linux/mlx5/cmd.h>
 #include <linux/module.h>
 #include "mlx5_core.h"
+#ifndef __GENKSYMS__
 #include "eswitch.h"
+#endif
 #include "../../mlxfw/mlxfw.h"
 
 static int mlx5_cmd_query_adapter(struct mlx5_core_dev *dev, u32 *out,
